pkg/services/ngalert/schedule/fetcher.go

```diff
@@ -46,21 +46,22 @@ func (sch *schedule) updateSchedulableAlertRules(ctx context.Context) error {
 	if !sch.schedulableAlertRules.isEmpty() {
 		keys, err := sch.ruleStore.GetAlertRulesKeysForScheduling(ctx)
 		if err != nil {
-			return err
+			return diff{}, err
 		}
 		if !sch.schedulableAlertRules.needsUpdate(keys) {
 			sch.log.Debug("No changes detected. Skip updating")
-			return nil
+			return diff{}, nil
 		}
+		// if diff is not empty then fetch ALL rules from the database.
```

Comment:
This comment is heavily misleading - I'd consider removing it.

There's nothing that returns a non empty diff before this comment.

Reply:
I suggested something below.

Reply:
This is still unresolved.

Reply:
yep. sorry I thought your suggestion below replaced it but it just added one below